docs/how-to-migrate-from-gazebo-classic.md

```diff
@@ -0,0 +1,146 @@
+# How to migrate from Gazebo Classic and `gazebo-yarp-plugins`
+
+## Introduction
+
+This brief guide will help you migrate an existing project created initially for Gazebo Classic simulator and `gazebo-yarp-plugins` to Modern Gazebo (`gz-sim`) and `gz-sim-yarp-plugins`.
+
+It is (usually) a quick and hassle-free process. In case you find errors or you need a feature missing from `gazebo-yarp-plugins` feel free to [open an issue](https://github.com/robotology/gz-sim-yarp-plugins/issues).
+
+The migration process usually entails the following main stages:
+
+1. Update of the robot model;
+1. Update of the plugin configuration;
+1. World set up;
+1. Environment variables;
+
+The next sections will address these steps in detail.
+
+## Update of the robot model
+
+Depending on the migration requirements and the model complexity, there are two ways to proceed with the migration: 
+
+- create a new robot model from scratch;
+- update the existing one used with Gazebo Classic.
+
+Both approaches work fine, in the latter case you end up with a single model that could be used with both simulators at the expense of having some harmless error messages at the startup of the simulation since some of the plugins will not be found.
+
+### gz-sim-yarp-plugins
+
+In the table below are described the `gz-sim-yarp-plugins` currently available, along with the `name` and `filename` attributes that have to be used to make Modern Gazebo load them.
+
+| Plugin          | `name` attribute         | `filename` attribute                |
+|-----------------|--------------------------|-------------------------------------|
+| Base state      | `gzyarp::BaseState`      | `gz-sim-yarp-basestate-system`      |
+| RGB Camera      | `gzyarp::Camera`         | `gz-sim-yarp-camera-system`         |
+| Clock           | `gzyarp::Clock`          | `gz-sim-yarp-clock-system`          |
+| Control board   | `gzyarp::ControlBoard`   | `gz-sim-yarp-controlboard-system`   |
+| Force-Torque    | `gzyarp::ForceTorque`    | `gz-sim-yarp-forcetorque-system`    |
+| IMU             | `gzyarp::Imu`            | `gz-sim-yarp-imu-system`            |
+| Laser           | `gzyarp::Laser`          | `gz-sim-yarp-laser-system`          |
+| Robot interface | `gzyarp::RobotInterface` | `gz-sim-yarp-robotinterface-system` |
```

Comment:
Could it make sense to have a table of correspondence between gazebo-yarp-plugins and gz-sim-yarp-plugins names? 

Probably I think it could also make sense to link PRs that migrated existing models from gazebo-yarp-plugins and gz-sim-yarp-plugins, such as https://github.com/icub-tech-iit/ergocub-software/pull/230 . Typically those are quite useful as they provide a reference in all possible complexity the process.

Reply:
Yes it has sense to me, but only concerning the `filename` attribute of `gazebo-yarp-plugins` since for Gazebo Classic the name is a custom name for each instance of the plugin, e.g.
```xml
<plugin name="controlboard_left_leg" filename="libgazebo_yarp_controlboard.so">
```

> Probably I think it could also make sense to link PRs that migrated existing models from gazebo-yarp-plugins and gz-sim-yarp-plugins, such as [icub-tech-iit/ergocub-software#230](https://github.com/icub-tech-iit/ergocub-software/pull/230) . Typically those are quite useful as they provide a reference in all possible complexity the process.

Yes absolutely!